main/lsp/LSPPreprocessor.cc

```diff
@@ -321,7 +321,7 @@ LSPPreprocessor::canonicalizeEdits(u4 v, unique_ptr<DidOpenTextDocumentParams> o
         string localPath = config->remoteName2Local(uri);
         if (!config->isFileIgnored(localPath)) {
             auto fileType = getFileType(localPath, config->opts);
-            auto file = make_shared<core::File>(move(localPath), move(openParams->textDocument->text), fileType, v);
+            auto file = make_shared<core::File>(string(localPath), move(openParams->textDocument->text), fileType, v);
```

Comment:
I guess it's not ok to move the string here and it needs to be duplicated? Is that because it's used later on in the same function?

Reply:
Yeah, it's used to index into `openFiles` a couple of lines later.

Though I guess `test/helpers/expectations.cc` solved this by grabbing a slot before moving the filename, and the same technique could be done here.  Probably more profitable to avoid an allocation in LSP code than it is in test code.  I'll do that tomorrow before merging.